gridcomps/ExtData3G/ExtDataGridComp.F90

```diff
@@ -144,6 +172,49 @@ subroutine run(gridcomp, importState, exportState, clock, rc)
       _RETURN(_SUCCESS)
    end subroutine run
 
+  function get_item_index(this,base_name,current_time,rc) result(item_index)
+     integer :: item_index
+     class(ExtDataGridComp), intent(in) :: this
+     type(ESMF_Time) :: current_time
+     character(len=*),intent(in) :: base_name
+     integer, optional, intent(out) :: rc
+
+     character(len=:), allocatable :: export_name
+     integer :: i
+     integer, pointer :: num_rules,i_start
+     logical :: found
+     type(PrimaryExport), pointer :: item
+
+     found = .false.
+     do i=1,this%export_vector%size()
+        item => this%export_vector%at(i)
+        export_name = item%get_export_var_name()
+        if (export_name == base_name) then
+           found = .true.
+           i_start => this%export_id_start%at(i)
+           num_rules => this%rules_per_export%at(i)
+           exit
+        end if
+     enddo
+     _ASSERT(found,"ExtData no item with basename '"//TRIM(base_name)//"' found")
+
+     item_index = -1
+     if (num_rules == 1) then
+        item_index = i_start
+     else if (num_rules > 1) then
```

Comment:
Is this conditional needed? Looks like the else block might work for both.

Reply:
yes I need the conditional, don't want to loop if there's only one rule, nothing to do and the starting times for an item won't even be defined if there's only one rule